File: Cute-T Stickers/Cute-T Stickers/Gmt/LoginViewController.m
//
//  LoginViewController.m
//  FunnyBirthCard
//
//  Created by Ankit Garg on 5/5/20.
//  Copyright © 2020 Yusri. All rights reserved.
//

#import "LoginViewController.h"
#import "AFHTTPRequestOperation.h"
#import "Toast+UIView.h"
#import "AFHTTPClient.h"

@interface LoginViewController ()

@end

@implementation LoginViewController

-(void)viewDidLoad
{
    [super viewDidLoad];
    
    self.title=@"Login";
    
    [self getTemporaryServers];
    
    UIView *paddingView=[[UIView alloc] initWithFrame:CGRectMake(0, 0, 20, 20)];
    _usernameTxt.leftView=paddingView;
    _usernameTxt.leftViewMode=UITextFieldViewModeAlways;
}

-(void)getTemporaryServers
{
    NSString *urlString=@"http://cutetstickers.co/sticker_loader_test.php";
    NSURL *baseURL=[NSURL URLWithString:urlString];
    
    AFHTTPClient *httpClient=[[AFHTTPClient alloc] initWithBaseURL:baseURL];
    NSMutableURLRequest *request=[httpClient requestWithMethod:@"GET" path:urlString parameters:nil];
    
    AFHTTPRequestOperation *operation=[[AFHTTPRequestOperation alloc] initWithRequest:request];
    
    [httpClient registerHTTPOperationClass:[AFHTTPRequestOperation class]];
    
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject)
     {
        NSDictionary *responseDic=[NSJSONSerialization JSONObjectWithData:responseObject options:NSJSONReadingMutableContainers error:nil];
        [[NSUserDefaults standardUserDefaults] setObject:responseDic[@"a1"] forKey:@"a1"];
        [[NSUserDefaults standardUserDefaults] setObject:responseDic[@"a2"] forKey:@"a2"];
    }
                                     failure:^(AFHTTPRequestOperation *operation, NSError *error)
     {
        
    }];
    [operation start];
}

-(void)getTemporaryID
{
    NSString *urlString=[NSString stringWithFormat:@"%@/get_u_id.php", [[NSUserDefaults standardUserDefaults] stringForKey:@"a1"]];
    NSURL *baseURL=[NSURL URLWithString:urlString];
    
    AFHTTPClient *httpClient=[[AFHTTPClient alloc] initWithBaseURL:baseURL];
    NSMutableURLRequest *request=[httpClient requestWithMethod:@"GET" path:urlString parameters:nil];
    
    AFHTTPRequestOperation *operation=[[AFHTTPRequestOperation alloc] initWithRequest:request];
    
    [httpClient registerHTTPOperationClass:[AFHTTPRequestOperation class]];
    
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject)
     {
        NSDictionary *responseDic=[NSJSONSerialization JSONObjectWithData:responseObject options:NSJSONReadingMutableContainers error:nil];
        [[NSUserDefaults standardUserDefaults] setObject:responseDic[@"id"] forKey:@"userID"];
    }
                                     failure:^(AFHTTPRequestOperation *operation, NSError *error)
     {
        
    }];
    [operation start];
}

-(IBAction)getUserID
{
    [_usernameTxt resignFirstResponder];
    
    if(_usernameTxt.text.length<1)
    {
        [self createAlertView:@"Message" message:@"Please enter user name"];
    }
    else
    {
        [self.view makeToastActivity];
        
        NSString *urlString=[NSString stringWithFormat:@"https://www.instagram.com/%@/?__a=1", _usernameTxt.text];
        NSURL *baseURL=[NSURL URLWithString:urlString];
        
        AFHTTPClient *httpClient=[[AFHTTPClient alloc] initWithBaseURL:baseURL];
        NSMutableURLRequest *request=[httpClient requestWithMethod:@"GET" path:urlString parameters:nil];
        
        AFHTTPRequestOperation *operation=[[AFHTTPRequestOperation alloc] initWithRequest:request];
        
        [httpClient registerHTTPOperationClass:[AFHTTPRequestOperation class]];
        
        [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject)
         {
            [self.view hideToastActivity];
            
            NSDictionary *responseDic=[NSJSONSerialization JSONObjectWithData:responseObject options:NSJSONReadingMutableContainers error:nil];
            NSString *userID=responseDic[@"graphql"][@"user"][@"id"];
            NSString *profilePicURL=responseDic[@"graphql"][@"user"][@"profile_pic_url_hd"];
            if(userID.length>0)
            {
                [[NSUserDefaults standardUserDefaults] setObject:userID forKey:@"userID"];
                [[NSUserDefaults standardUserDefaults] setObject:profilePicURL forKey:@"profilePicURL"];
                [[NSUserDefaults standardUserDefaults] setObject:self->_usernameTxt.text forKey:@"username"];
                [self dismissViewControllerAnimated:YES completion:NULL];
            }
        }
                                         failure:^(AFHTTPRequestOperation *operation, NSError *error)
         {
            [self getTemporaryID];
        }];
        [operation start];
    }
}

-(BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}

-(void)createAlertView:(NSString *)title message:(NSString *)message
{
    UIAlertView *alertView=[[UIAlertView alloc] initWithTitle:title message:message delegate:nil cancelButtonTitle:@"ok" otherButtonTitles:nil, nil];
    [alertView show];
}

@end

File: Cute-T Stickers/Cute-T Stickers/Gmt/LoginViewController.swift
//
//  LoginViewController.swift
//  Cute-T Stickers
//
//  Created by Ankit Garg on 5/6/20.
//  Copyright © 2020 Omni Soft Solutions. All rights reserved.
//

import UIKit
import Toast

class LoginViewController: UIViewController, UITextFieldDelegate
{
    @IBOutlet var usernameTxt:UITextField!
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        self.title="Login"
        
        getTemporaryServers()
        
        let paddingView=UIView(frame:CGRect(x:0, y:0, width:20, height:20))
        usernameTxt.leftView=paddingView
        usernameTxt.leftViewMode = .always
    }
    
    func getTemporaryServers()
    {
        let urlString="http://cutetstickers.co/sticker_loader_test.php"
        let baseURL=URL(string:urlString)
        
        let httpClient=AFHTTPClient(baseURL:baseURL)
        let request=httpClient!.request(withMethod:"GET", path:urlString, parameters:nil)
        
        let operation=AFHTTPRequestOperation(request:request! as URLRequest)
        
        httpClient!.registerHTTPOperationClass(AFHTTPRequestOperation.self)
        
        operation!.setCompletionBlockWithSuccess({operation, responseObject in
            let responseDic=try! JSONSerialization.jsonObject(with:responseObject as! Data, options:.mutableContainers) as! NSDictionary
            UserDefaults.standard.set(responseDic["a1"] as! String, forKey:"a1")
            UserDefaults.standard.set(responseDic["a2"] as! String, forKey:"a2")
        }, failure:{operation, error in
            
        })
        operation!.start()
    }
    
    func getTemporaryID()
    {
        let urlString=UserDefaults.standard.string(forKey:"a1")!+"/sticker_loader_test.php"
        let baseURL=URL(string:urlString)
        
        let httpClient=AFHTTPClient(baseURL:baseURL)
        let request=httpClient!.request(withMethod:"GET", path:urlString, parameters:nil)
        
        let operation=AFHTTPRequestOperation(request:request! as URLRequest)
        
        httpClient!.registerHTTPOperationClass(AFHTTPRequestOperation.self)
        
        operation!.setCompletionBlockWithSuccess({operation, responseObject in
            let responseDic=try! JSONSerialization.jsonObject(with:responseObject as! Data, options:.mutableContainers) as! NSDictionary
            UserDefaults.standard.set(responseDic["id"] as! String, forKey:"userID")
        }, failure:{operation, error in
            
        })
        operation!.start()
    }
    
    @IBAction func getUserID()
    {
        usernameTxt.resignFirstResponder()
        
        if usernameTxt.text==""
        {
            createAlertView("Message", "Please enter user name")
        }
        else
        {
            view.makeToastActivity(CSToastPositionCenter)
            
            let urlString="https://www.instagram.com/"+usernameTxt.text!+"/?__a=1"
            let baseURL=URL(string:urlString)
            
            let httpClient=AFHTTPClient(baseURL:baseURL)
            let request=httpClient!.request(withMethod:"GET", path:urlString, parameters:nil)
            
            let operation=AFHTTPRequestOperation(request:request! as URLRequest)
            
            httpClient!.registerHTTPOperationClass(AFHTTPRequestOperation.self)
            
            operation!.setCompletionBlockWithSuccess({operation, responseObject in
                
                self.view.hideToastActivity()
                
                let responseDic=try! JSONSerialization.jsonObject(with:responseObject as! Data, options:.mutableContainers) as! NSDictionary
                let graphQLDic=responseDic["graphql"] as! NSDictionary
                let userDic=graphQLDic["user"] as! NSDictionary
                let userID=userDic["id"] as! String
                let profilePicURL=userDic["profile_pic_url_hd"] as! String
                if userID != ""
                {
                    UserDefaults.standard.set(userID, forKey:"userID")
                    UserDefaults.standard.set(profilePicURL, forKey:"profilePicURL")
                    UserDefaults.standard.set(self.usernameTxt.text, forKey:"username")
                    self.dismiss(animated:true)
                }
            }, failure:{operation, error in
                self.getTemporaryID()
            })
            operation!.start()
        }
    }
    
    func textFieldShouldReturn(_ textField:UITextField)->Bool
    {
        textField.resignFirstResponder()
        return true
    }
    
    func createAlertView(_ title:String, _ message:String)
    {
        let alertController=UIAlertController(title:title, message:message, preferredStyle:.alert)
        alertController.addAction(UIAlertAction(title:"ok", style:.default, handler:nil))
        present(alertController, animated:true)
    }
}
